File: PDS_E-Market_Equipo03-branch-nacho/src/main/java/com/emarket/service/interfaz/AuthService.java
package com.emarket.service.interfaz;

import com.emarket.dto.auth.LoginRequestDto;
import com.emarket.dto.auth.LoginResponseDto;
import com.emarket.entity.Permiso;
import com.emarket.entity.Usuario;

public interface AuthService {
    LoginResponseDto login(LoginRequestDto dto);
    void logout(String token);
    Usuario obtenerUsuarioAutenticado(String token);
    Usuario validarPermiso(String token, Permiso permiso);
    Usuario validarEsAdministrador(String token);
}

File: frontend/src/services/auth.service.ts
import { apiClient } from '../core/api/axiosInstance';
import type {
  LoginRequestDto,
  LoginResponseDto,
  PerfilResponse,
} from '../core/types';

/**
 * Servicio de autenticación: login, logout y perfil.
 */
export const authService = {
  async login(dto: LoginRequestDto): Promise<LoginResponseDto> {
    const { data } = await apiClient.post<LoginResponseDto>('/auth/login', dto);
    return data;
  },

  async logout(): Promise<void> {
    await apiClient.post('/auth/logout');
  },

  async getPerfil(): Promise<PerfilResponse> {
    const { data } = await apiClient.get<PerfilResponse>('/auth/perfil');
    return data;
  },
};
